docs/examples/Experimental.tsx

```diff
@@ -183,73 +183,47 @@ const Option = (props: OptionProps<DateOption, false>) => {
 
 interface DatePickerProps {
   readonly value: DateOption | null;
-  readonly onChange: (value: DateOption | null) => void;
+  readonly onChange: (newValue: DateOption | null) => void;
 }
 
-interface DatePickerState {
-  readonly options: readonly (DateOption | CalendarGroup)[];
-}
+const DatePicker = (props: DatePickerProps) => {
+  const [options, setOptions] = useState(defaultOptions);
 
-class DatePicker extends Component<DatePickerProps, DatePickerState> {
-  state: DatePickerState = {
-    options: defaultOptions,
-  };
-  handleInputChange = (value: string) => {
+  const handleInputChange = (value: string) => {
     if (!value) {
-      this.setState({ options: defaultOptions });
+      setOptions(defaultOptions);
       return;
     }
     const date = chrono.parseDate(suggest(value.toLowerCase()));
-    if (date) {
-      this.setState({
-        options: [createOptionForDate(date), createCalendarOptions(date)],
-      });
-    } else {
-      this.setState({
-        options: [],
-      });
+    if (!date) {
+      setOptions([]);
+      return;
     }
+    setOptions([createOptionForDate(date), createCalendarOptions(date)]);
   };
-  render() {
-    const { value } = this.props;
-    const { options } = this.state;
-    return (
-      <Select<DateOption, false>
-        {...this.props}
-        components={{ Group, Option }}
-        filterOption={null}
-        isMulti={false}
-        isOptionSelected={(o, v) => v.some((i) => i.date.isSame(o.date, 'day'))}
-        maxMenuHeight={380}
-        onChange={this.props.onChange}
-        onInputChange={this.handleInputChange}
-        options={options}
-        value={value}
-      />
-    );
-  }
-}
 
-interface State {
-  readonly value: DateOption | null;
-}
+  return (
+    <Select<DateOption, false>
+      {...props}
+      components={{ Group, Option }}
+      filterOption={null}
+      isMulti={false}
+      isOptionSelected={(o, v) => v.some((i) => i.date.isSame(o.date, 'day'))}
+      maxMenuHeight={380}
+      onChange={props.onChange}
+      onInputChange={handleInputChange}
+      options={options}
+      value={props.value}
+    />
+  );
+};
 
-export default class Experimental extends Component<{}, State> {
-  state: State = {
-    value: defaultOptions[0] as DateOption,
-  };
-  handleChange = (value: DateOption | null) => {
-    this.setState({ value });
-  };
-  render() {
-    const { value } = this.state;
-    const displayValue =
```

Comment:
I've removed this to simplify the example.

Reply:
I kinda like it for this example ¯\_(ツ)_/¯
`value.value.toString()` is a bit grim though — maybe `const [dateOption, setDateOption] = useState<DateOption | null>(defaultOptions[0] as DateOption)` would be better?

Reply:
🤔 Hm. Isn't that what i'm already doing in [line 222](https://github.com/JedWatson/react-select/pull/5393/files#diff-b0383a87b0483e38ff709daf238f585c7c7d5cac514b2bd81d1a6ae4c9b08ce4R222) (except i've just left it named as `value` instead of renaming it to `dateOption`)?